datasquare.go

```diff
@@ -6,7 +6,8 @@ import (
 )
 
 type dataSquare struct {
-	square       [][][]byte
+	squareRow    [][][]byte // row-major
+	squareCol    [][][]byte // col-major
```

Comment:
Not sure how I feel about doubling the memory footprint here. Also, this makes the implementation more error-prone. Are you sure the other approach "would be an astronomical amount of work"?

Reply:
Interesting that the number of allocs doesn't really seem to increase though 🤔

Reply:
Doubling memory footprint shouldn't be too bad. 1MB -> 2MB. That's nothing (ofc I'm biased with a 128 GB machine, but you get the idea). The real time saver is in _number of allocations_, not size of allocations. RAM is meant to be used and unused RAM is wasted RAM.

The codec libraries for example assume a `[][]byte` input, not an EDS data structure. So we'd have to fundamentally change all those interfaces.

Reply:
Makes sense. I would somehow have expected that the _number_ of allocs would increase more significantly.

Thanks for the detailed reply. I agree with this approach now.

Reply:
> Interesting that the number of allocs doesn't really seem to increase though

The data squares are allocated once, and `columnSlice` is now zero-allocation instead of requiring at least one allocation per call. So not so surprising.